etl/parse_email_save_attachment/parse_messages.py

```diff
@@ -0,0 +1,127 @@
+#!/usr/bin/env python3
+
+import os
+import pytz
+import boto3
+import email
+import tempfile
+from email import policy
+from operator import itemgetter
+from datetime import datetime
+
+
+def get_most_recent_file(bucket_name, prefix):
+    s3 = boto3.client("s3")
+
+    # List objects within the specified bucket and prefix
+    response = s3.list_objects_v2(Bucket=bucket_name, Prefix=prefix)
+
+    # Check if the bucket is empty or the prefix doesn't match any files
+    if "Contents" not in response:
+        return "No files found in the bucket with the specified prefix."
+
+    # Sort the files by last modified date
+    files = sorted(response["Contents"], key=itemgetter("LastModified"), reverse=True)
+
+    # Get the most recent file
+    most_recent_file = files[0]
+    return most_recent_file["Key"]
+
+
+def get_file_content(bucket_name, file_key):
+    s3 = boto3.client("s3")
+
+    # Get the object
+    obj = s3.get_object(Bucket=bucket_name, Key=file_key)
+
+    # Get the object's content
+    content = obj["Body"].read().decode("utf-8")
+    return content
+
+
+def parse_email_from_s3(email_content):
```

Comment:
this function might benefit from a docstring that summarizes what it does and what it returns

Reply:
You're so right. Thank you for this suggestion. I have added [some docstrings](https://github.com/cityofaustin/dts-maximo-geo-integration/pull/1/commits/c31f1c985e91ed1747b242e6d40e81ac41cc1ce1) for all the functions. I definitely leaned on copilot to help with this, but I've checked them each and they are accurate. I'd be eager to work on any feedback you have though, because I am a novice-at-best when it comes to documentation.